examples/quickstart_examples/simple_three_step_cursor/connector.py

```diff
@@ -42,12 +42,16 @@ def schema(configuration: dict):
 # - state: a dictionary contains whatever state you have chosen to checkpoint during the prior sync
 # The state dictionary is empty for the first sync or for any full re-sync
 def update(configuration: dict, state: dict):
+    log.warning("Example: QuickStart Examples - Simple Three Step Cursor")
+
     # Retrieve the cursor from the state object to determine the current position in the SOURCE_DATA.
     # If the cursor is not present in the state, start from the beginning (cursor = 0).
     cursor = state['cursor'] if 'cursor' in state else 0
     log.fine(f"current cursor is {repr(cursor)}")
 
     # Get the row of data from SOURCE_DATA using the cursor position.
+    if cursor >= SOURCE_DATA.__len__():
```

Comment:
Do we want to throw an exception? Otherwise, we can take the modulus of the cursor by 3 and return the related data at that index.

Reply:
we want the connector to break and stop after 3 iterations, otherwise it will run forever.